src/System.Linq.Expressions/src/System/Linq/Expressions/Interpreter/LightLambda.cs

```diff
@@ -33,6 +34,175 @@ internal LightLambda(LightDelegateCreator delegateCreator, IStrongBox[] closure)
+                int count;
+                if (!_tryStart.TryGetValue(index, out count))
+                {
+                    _tryStart.Add(index, 1);
+                    return;
+                }
+
+                _tryStart[index] = count + 1;
+            }
+
+            private void AddHandlerExit(int index)
+            {
+                int count;
+                if (!_handlerExit.TryGetValue(index, out count))
+                {
+                    _handlerExit.Add(index, 1);
+                    return;
+                }
+
+                _handlerExit[index] = count + 1;
+            }
+
+            private void Indent()
+            {
+                _indent = new string(' ', _indent.Length + 2);
+            }
+
+            private void Dedent()
+            {
+                _indent = new string(' ', _indent.Length - 2);
+            }
+
+            public override string ToString()
+            {
+                var sb = new StringBuilder();
+
+                string name = _interpreter.Name ?? "lambda_method";
+                sb.AppendLine("object " + name + "(object[])");
+                sb.AppendLine("{");
+
+                sb.AppendLine("  .locals " + _interpreter.LocalCount);
+                sb.AppendLine("  .maxstack " + _interpreter.Instructions.MaxStackDepth);
+                sb.AppendLine("  .maxcontinuation " + _interpreter.Instructions.MaxContinuationDepth);
+                sb.AppendLine();
+
+                Instruction[] instructions = _interpreter.Instructions.Instructions;
+                InstructionArray.DebugView debugView = new InstructionArray.DebugView(_interpreter.Instructions);
+                InstructionList.DebugView.InstructionView[] instructionViews = debugView.GetInstructionViews(includeDebugCookies: false);
+
+                for (int i = 0; i < instructions.Length; i++)
+                {
+                    EmitExits(sb, i);
+
+                    int startCount;
+                    if (_tryStart.TryGetValue(i, out startCount))
+                    {
+                        for (int j = 0; j < startCount; j++)
+                        {
+                            sb.AppendLine(_indent + ".try");
+                            sb.AppendLine(_indent + "{");
+                            Indent();
+                        }
+                    }
+
+                    string handler;
+                    if (_handlerEnter.TryGetValue(i, out handler))
+                    {
+                        sb.AppendLine(_indent + handler);
+                        sb.AppendLine(_indent + "{");
+                        Indent();
+                    }
+
+                    Instruction instruction = instructions[i];
+                    InstructionList.DebugView.InstructionView instructionView = instructionViews[i];
+
+                    sb.AppendLine(_indent + string.Format(CultureInfo.InvariantCulture, "IP_{0}: {1}", i.ToString().PadLeft(4, '0'), instructionView.GetValue()));
```

Comment:
Perf aside, you might want to include the indent in the format string anyway, e.g.
```C#
sb.AppendFormat(CultureInfo.InvariantCulture, "{0}IP_{1}: {2}", _indent, i.ToString().PadLeft(4, '0'), instructionView.GetValue()).AppendLine();
```
```